dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/ParserLitmusAArch64.java

```diff
@@ -21,12 +20,10 @@ public Program parse(CharStream charStream) {
         LitmusAArch64Parser parser = new LitmusAArch64Parser(tokenStream);
         parser.addErrorListener(new DiagnosticErrorListener(true));
         parser.addErrorListener(new ParserErrorListener());
-        ProgramBuilder pb = new ProgramBuilder(SourceLanguage.LITMUS);
         ParserRuleContext parserEntryPoint = parser.main();
-        VisitorLitmusAArch64 visitor = new VisitorLitmusAArch64(pb);
+        VisitorLitmusAArch64 visitor = new VisitorLitmusAArch64();
 
         Program program = (Program) parserEntryPoint.accept(visitor);
-        program.setArch(Arch.ARM8);
```

Comment:
This clashes with #480.

I just saw `ProgramBuilder.forArch()` method. Just to be sure, they arch is eventually set for litmus tests, right?

Reply:
It is actually set immediately inside `ProgramBuilder.forArch()`. `ProgramBuilder` now starts from an empty `Program` instance and adds functions/threads/events to that instance.